ocp_resources/performance_profile.py

```diff
@@ -0,0 +1,112 @@
+            machine_config_pool_selector (str): Defines the MachineConfigPool label to use in the
+                MachineConfigPoolSelector of resources like KubeletConfigs created by the operator.
+            net (dict): Defines a set of network related features. Expected key-values are:
+                devices: (dict). Expected key-values are:
+                    deviceID (str): Network device ID represented by 16 digit hexadecimal number.
+                    interfaceName (str): Network device name to be matched.
+                    vendorID (str): Network device vendor ID represented by 16 digit hexadecimal number.
+                userLevelNetworking (boolean, default: False): when enabled sets either all or specified network devices queue size to the amount of reserved CPUs.
+            node_selector (str, required): Defines the node label to use in the NodeSelectors of resources like Tuned created by the operator.
+            numa (dict): Defines options related to topology aware affinities. Expected key-values are:
+                topologyPolicy (str): Name of the policy when TopologyManager is enabled.
+            real_time_kernel (dict):  Defines a set of real time kernel related parameters. Expected key-values are:
+                enabled (boolean, default: False): Defines if the real time kernel packages should be installed.
+            workload_hints (dict): Defines hints for different types of workloads. Expected key-values are:
+                highPowerConsumption (boolean, default: False): Defines if the node should be configured for high power consumption.
+                mixedCpus (boolean, default: False): Enables mixed-cpu-node-plugin on the node if set.
+                perPodPowerManagement (boolean, default: False): Defines if the node should be configured for per pod power management.
+                realTime (boolean, default: True): Defines if the node should be configured for real time workload.
+        """
+
+        super().__init__(**kwargs)
+        self.additional_kernel_args = additional_kernel_args
+        self.cpu = cpu
+        self.globally_disable_irq_load_balancing = globally_disable_irq_load_balancing
+        self.hugepages = hugepages
+        self.machine_config_label = machine_config_label
+        self.machine_config_pool_selector = machine_config_pool_selector
+        self.net = net
+        self.node_selector = node_selector
+        self.numa = numa
+        self.real_time_kernel = real_time_kernel
+        self.workload_hints = workload_hints
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
+            manifest_spec = {}
+            if self.additional_kernel_args:
+                manifest_spec["additionalKernelArgs"] = self.additional_kernel_args
+
+            if self.cpu:
+                manifest_spec["cpu"] = self.cpu
+
+            if self.globally_disable_irq_load_balancing:
```

Comment:
You need to check that it is not None, `False` also match this `if`

Reply:
done